opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -102,21 +119,39 @@ def __init__(
         self.description = description
         self.unit = unit
         self.value_type = value_type
+        self.meter = meter
         self.label_keys = label_keys
         self.enabled = enabled
         self.monotonic = monotonic
         self.handles = {}
 
-    def get_handle(self, label_values: Sequence[str]) -> BaseHandle:
+    def get_handle(self, label_set: LabelSet) -> BaseHandle:
         """See `opentelemetry.metrics.Metric.get_handle`."""
-        handle = self.handles.get(label_values)
+        lable_set_for = self._label_set_for(label_set)
+        handle = self.handles.get(lable_set_for.encoded)
         if not handle:
             handle = self.HANDLE_TYPE(
                 self.value_type, self.enabled, self.monotonic
             )
-        self.handles[label_values] = handle
+        self.handles[lable_set_for.encoded] = handle
         return handle
 
+    def _label_set_for(self, label_set: LabelSet) -> LabelSet:
+        """Returns an appropriate `LabelSet` based off this metric's `meter`
+
+        If this metric's `meter` is the same as label_set's meter, that means
+        label_set was created from this metric's `meter` instance and the
+        proper handle can be found. If not, label_set was created using a
+        different `meter` implementation, in which the metric cannot interpret.
+        In this case, return the `EMPTY_LABEL_SET`.
+
+        Args:
+            label_set: The `LabelSet` to check for the same `Meter` implentation.
+        """
+        if label_set.meter and label_set.meter is self.meter:
+            return label_set
+        return EMPTY_LABEL_SET
```

Comment:
This seems like a pretty dangerous default. If the user passes in a labelset created by a different meter, we hand back the (real) handle for the empty/default labelset? 

Shouldn't `get_handle` return a no-op handle in that case (and log a warning) instead?